msm/src/mvlookup.rs

```diff
@@ -0,0 +1,171 @@
+//! Implement the protocol MVLookup <https://eprint.iacr.org/2022/1530.pdf>
+
+use std::iter;
+
+use ark_ff::{FftField, Field};
+use kimchi::circuits::domains::EvaluationDomains;
+use rand::{seq::SliceRandom, thread_rng, Rng};
+
+// TODO: Add more built-in lookup tables
+#[derive(Copy, Clone, Debug)]
+pub enum LookupTable {
```

Comment:
Potential suggestion for the future: if possible let's make it generic, e.g. instead of `enum` implement a trait and then pass concretely used separately defined tables into the prover.

Reply:
Opened follow-up https://github.com/o1-labs/proof-systems/issues/1824